### PR TITLE
feat(signing): add window arweave wallet to available signing options

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: src/token.ts [hunk 1]
```diff
@@ -153,3 +153,10 @@ export class mIOToken extends PositiveFiniteInteger {
     return new IOToken(this.valueOf() / MIO_PER_IO);
   }
 }
+
+export type AoSigningFuction = (args: {
```
Comment (kunstmusik) [inline src/token.ts:157]: Perhaps this should be called AoSigner, considering it's returned from createAoSigner()

File: src/common/ant.ts [hunk 2]
```diff
@@ -227,14 +227,14 @@ export class AoANTReadable implements AoANTRead {
 }
 
 export class AoANTWriteable extends AoANTReadable implements AoANTWrite {
-  private signer: ContractSigner;
+  private signer: AoSigningFuction;
```
Comment (kunstmusik) [inline src/common/ant.ts:230]: might be good to rename "signer" to "aoSigner" through the SDK to make it self-documenting

File: examples/esm/index.mjs [hunk 3]
```diff
@@ -66,10 +68,14 @@ import Arweave from 'arweave';
...
 
   const jwk = await arweave.wallets.generate();
-
-  const processId = await spawnANT({
-    signer: new ArweaveSigner(jwk),
-  });
+  let processId;
+  try {
+    processId = await spawnANT({
+      signer: AOProcess.createAoSigner(new ArweaveSigner(jwk)),
```
Comment (dtfiedler) [inline examples/esm/index.mjs:74]: i'm in favor of moving `createAoSigner` to a standalone utility function that we export and use in the class. there's no need for it to be static anymore

**Review by dtfiedler — CHANGES_REQUESTED** [review]
see comment r.e. moving `createAoSigner` to a utility function

File: src/utils/ao.ts [hunk 4]
```diff
@@ -120,3 +121,35 @@ export async function evolveANT({
...
 }
+
+export function createAoSigner(
+  signer: ContractSigner,
+): (args: {
+  data: string | Buffer;
+  tags?: { name: string; value: string }[];
+  target?: string;
+  anchor?: string;
+}) => Promise<{ id: string; raw: ArrayBuffer }> {
```
Comment (dtfiedler) [inline src/utils/ao.ts:132]: update the type here now that we have `AoSigner`

File: examples/esm/index.mjs (hunk 3 above)
Comment (atticusofsparta) [reply examples/esm/index.mjs]: https://github.com/ar-io/ar-io-sdk/pull/175/commits/8e95eddf5723b0fdbf870688556f76a3e80bb6fa

File: src/utils/ao.ts (hunk 4 above)
Comment (atticusofsparta) [reply src/utils/ao.ts]: https://github.com/ar-io/ar-io-sdk/pull/175/commits/8e95eddf5723b0fdbf870688556f76a3e80bb6fa

File: package.json [hunk 5]
```diff
@@ -118,8 +118,9 @@
     "vite-plugin-node-polyfills": "^0.22.0"
   },
   "dependencies": {
-    "@permaweb/aoconnect": "^0.0.55",
+    "@permaweb/aoconnect": "^0.0.57",
     "arbundles": "0.11.0",
+    "arconnect": "^1.0.3",
```
Comment (dtfiedler) [inline package.json:123]: is this being used? if not required, remove from the build or put as a dev dep

Comment (atticusofsparta) [reply package.json:123]: This is the types package for arconnect, we do use it - will try as a dev dep

File: src/common/ant.ts (hunk 2 above)
Comment (dtfiedler) [reply src/common/ant.ts]: i prefer `signer` to keep it abstracted from it's type, but a nit

File: package.json (hunk 5 above)
Comment (atticusofsparta) [reply package.json]: https://github.com/ar-io/ar-io-sdk/pull/175/commits/34f07d25024c080f529218569cba57041e34baa2

Comment (dtfiedler) [reply package.json]: should be fine as a dev dep if we are not explicitly exporting any arconnect types from our own build

File: src/common/io.ts [hunk 6]
```diff
@@ -578,7 +579,7 @@ export class IOReadable implements AoIORead {
 
 export class IOWriteable extends IOReadable implements AoIOWrite {
   protected declare process: AOProcess;
-  private signer: ContractSigner;
+  private aoSigner: AoSigner;
```
Comment (dtfiedler) [inline src/common/io.ts:582]: going to ask to revert just to `signer` - should it ever change we don't have to modify the attribute name

Comment (atticusofsparta) [reply src/common/io.ts]: [3b23f80](https://github.com/ar-io/ar-io-sdk/pull/175/commits/3b23f80581ea331b89ac31756421f74799315e2c)

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**